File: javascript/simple-cipher/simple_cipher.rb
class Cipher

  ALPHABET = ('a'..'z').to_a

  def original_encode(message, offset)
    output = []
    message.chars.each do |letter|
      new_index = ALPHABET.index(letter) + offset
      output << ALPHABET[new_index]
    end
    output.join()
  end

  def processor(message, offset, adjusted_index_proc)
    output = []
    message.chars.each do |letter|
      new_index = adjusted_index_proc.call(letter)
      output << ALPHABET[new_index]
    end
    output.join()
  end

  def encode(message, offset)
    proc = Proc.new { |letter| ALPHABET.index(letter) + offset }
    processor(message, offset, proc)
  end

  def decode(message, offset)
    proc = Proc.new { |letter| ALPHABET.index(letter) - offset }
    processor(message, offset, proc)
  end

end

cipher = Cipher.new

p cipher.original_encode('ilikekitties', 3)
p cipher.encode('ilikekitties', 3)

p cipher.decode('lolnhnlwwlhv', 3)

File: javascript/simple-cipher/simple-cipher.js
const alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];

class Cipher {
  constructor(cipherKey) {
    if(cipherKey != undefined && cipherKey === cipherKey.toUpperCase()) {
      throw Error('Bad key');
    }
    this.cupherKey = cipherKey
    this.key = alphabet.join('')

  }

  isUpperCase(str){
    return (/[A-Z]/.test(str));
  }

  encode(input, offset) {
    input = input.split('');
    var encoded = [];
    var alphaIndex = 0
    input.forEach(function(letter){
      alphaIndex = alphabet.indexOf(letter) + offset;
      encoded.push(alphabet[alphaIndex]);
    })
    console.log( encoded.join('') );
  }

  decode(input) {

  }

}


module.exports = Cipher;
//["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]

// var x = new Cipher();
// x.encode('iamapandabear', 3);

